File: add/codesnippet/VisualBasic/p-system.componentmodel._51_1.vb
    Private Sub PrintIndexItem2()
        ' Creates a new collection and assigns it the events for button1.
        Dim events As EventDescriptorCollection = TypeDescriptor.GetEvents(button1)
        
        ' Sets an EventDescriptor to the specific event.
        Dim myEvent As EventDescriptor = events("KeyDown")
        
        ' Prints the name of the event.
        textBox1.Text = myEvent.Name
    End Sub 'PrintIndexItem2

File: add/codesnippet/CSharp/p-system.componentmodel._51_1.cs
private void PrintIndexItem2() {
    // Creates a new collection and assigns it the events for button1.
    EventDescriptorCollection events = TypeDescriptor.GetEvents(button1);
 
    // Sets an EventDescriptor to the specific event.
    EventDescriptor myEvent = events["KeyDown"];
 
    // Prints the name of the event.
    textBox1.Text = myEvent.Name;
 }
